federation/pkg/federation-controller/service/dns.go

```diff
@@ -170,10 +170,12 @@ func (s *ServiceController) ensureDnsRrsets(dnsZoneName, dnsName string, endpoin
 		if len(endpoints) < 1 {
 			glog.V(4).Infof("There are no healthy endpoint addresses at level %q, so CNAME to %q, if provided", dnsName, uplevelCname)
 			if uplevelCname != "" {
+				cs := rrsets.StartChangeset()
```

Comment:
Done, except that where I didn't need a variable I used the fluent style e.g.:

`err = rrsets.StartChangeset().Add(newRrset).Remove(oldRrset).Apply()`

Partially because I think it demonstrates why I don't think we need a simpler interface!